testnet-client/src/main.rs

```diff
@@ -358,23 +361,27 @@ string that we send you. It should look something like:
 
         // Read and parse B58 request code.
         #[derive(Clone)]
-        struct WrappedRequestPayload(pub Option<RequestPayload>);
+        struct WrappedRequestPayload(pub Option<PaymentRequest>);
         impl FromStr for WrappedRequestPayload {
             type Err = String;
             fn from_str(src: &str) -> Result<Self, Self::Err> {
                 if src.is_empty() {
                     return Ok(Self(None));
                 }
-
-                Ok(Self(Some(RequestPayload::decode(src).map_err(|err| {
-                    format!("Invalid request code: {}", err)
-                })?)))
+                let request_wrapper = PrintableWrapper::b58_decode(src.to_string())
+                    .map_err(|err| format!("Invalid request code: {}", err))?;
+                if !request_wrapper.has_payment_request() {
+                    return Err("Not a payment request code".to_string());
```

Comment:
I didn't think about this in the previous round, but I think we want to try and parse public addresses here and return a payment request with zero amount/no memo because I imagine users would be using them interchangeably. Otherwise the client won't let you pay an address.

Reply:
Now either will work, it's a nice feature, thanks for noticing!